tests/interface.bats

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env bats
```

Comment:
TIL about BATS, nice!

Reply:
BATS is amazing, and does a great job of improving the maintainability of Bash scripts. It does have a learning curve, but the payoff for scripts like this is worth it. It significantly decreases the time spend in the edit/test loop over pushing and running a build.